compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -169,22 +169,20 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             return;
         }
 
-        let size_in_bytes = src.layout.size.bytes();
-        if size_in_bytes == 0 {
-            // Nothing to write
```

Comment:
Is there a particular reason to remove the ZST check?

Reply:
Because, as I dug more, it was really only useful for my manual `memcpy`.  `OperandValue::store` already does it:

https://github.com/rust-lang/rust/blob/be8e5ba157a4ae494f9b4219b3b848e44ea5e8d3/compiler/rustc_codegen_ssa/src/mir/operand.rs#L285-L289